### PR TITLE
Added GNU Make cheatshet

### PR DESCRIPTION
**Pull request by OscarMZ**
IA page:

https://duck.co/ia/view/gnu_make_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by OscarMZ** [comment]
@moollaza: There is a problem with the IA page link. The id of the cheatsheet is gnu_make_cheat_sheet, but I the IA page is gnu_make_cheatsheet. The problem is that the test fails if I put gnu_make_cheatsheet as id, so I think the solution is to update the IA page to gnu_make_cheat_sheet.

Thanks for all!

File: share/goodie/cheat_sheets/json/gnu-make.json [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        "make", "make gnu", "makefile"
+    ],
+
+    "template_type": "terminal",
+
+    "section_order": [
+        "Make command line options",
+        "Automatic variables",
+        "Text manipulation functions",
+        "Variable assignment"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/gnu-make.json]: Lets do each word as snake-case for the sections (e.g., `Variable Assignment`).

File: share/goodie/cheat_sheets/json/gnu-make.json [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    },
+
+    "aliases": [
+        "make", "make gnu", "makefile"
+    ],
+
+    "template_type": "terminal",
+
+    "section_order": [
+        "Make command line options",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/gnu-make.json]: `Command-line Options`

File: share/goodie/cheat_sheets/json/gnu-make.json [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        "Automatic variables",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/gnu-make.json]: `Default Variables`

File: share/goodie/cheat_sheets/json/gnu-make.json [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            },
+            {
+                "key": "$^",
+                "val": "Names of all prerequisites"
+            }
+        ],
+        "Text-manipulation functions": [
+            {
+                "key": "$(subst from,to,text)",
+                "val": "Substitute substring from to to in text"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/gnu-make.json]: `from to to`?

Comment (OscarMZ) [inline share/goodie/cheat_sheets/json/gnu-make.json]: Yes, It's strange, but it's correct: http://www.cheatography.com/bavo-van-achte/cheat-sheets/gnumake/

Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/gnu-make.json]: Lets re-word it - I don't think it even makes sense.

Comment (moollaza) [inline share/goodie/cheat_sheets/json/gnu-make.json]: _replace from with to in text_

File: share/goodie/cheat_sheets/json/gnu-make.json [hunk 5]
```diff
@@ -0,0 +1,114 @@
+{
+    "id": "gnu_make_cheat_sheet",
+    "name": "Make Cheatsheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/gnu-make.json]: "Gnu Make" -- let's remove Cheatsheetq

File: share/goodie/cheat_sheets/json/gnu-make.json [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    "description": "A cheatsheet about GNU make",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/gnu-make.json]: Let's make this more descriptive. I suspect they know it’s a Gnu cheat sheet ;)

**Comment by moollaza** [comment]
@OscarMZ Thanks! This LGTM :+1:

Please update the text for title, description and the text substitution example we discussed above.

Without text formatting or other visual indicators reader wont know what it means.

**Comment by tagawa** [comment]
@OscarMZ, thanks again for creating the GNU Make cheat sheet! You’ve done a really good job here and we'd like you to help make DuckDuckGo even better for yourself and others interested in programming tools.

When you're ready to move on, please:

- Create a new Instant Answer for searches you do related to programming tools

OR

- Test live programming tools Instant Answers (or those in development) for bugs and suggest improvements

Detailed instructions explaining how you can help are available at the top of this [page](https://github.com/duckduckgo/duckduckgo/wiki/Creating-or-Improving-Instant-Answers).

Thanks again!

**Pull request merged**